utils/gen_offsets.py

```diff
@@ -0,0 +1,87 @@
+"""
+Utility script to generate offset definitions for a peripheral from an SVD file.
+
+Usage:
+
+    python gen_offsets.py \
+        --svd <svd_file> \
+        --output <output_file> \
+        --peripheral <peripheral_name> \
+        --guard <header_guard> \
+        [--prefix <prefix>]
+
+Copyright (c) 2024 Nordic Semiconductor ASA
+SPDX-License-Identifier: Apache-2.0
+"""
+
+import argparse
+from datetime import datetime
+from pathlib import Path
+import sys
+import xml.etree.ElementTree as ET
+
+
+HEADER = """/*
+ * Copyright (c) {} Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
```

Comment:
Maybe include a notice saying "this file is auto-generated".

Reply:
done